sentry/src/test/java/io/sentry/SentryContextTest.kt

```diff
@@ -0,0 +1,134 @@
+package io.sentry
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.slf4j.MDCContext
+import kotlinx.coroutines.withContext
+import org.slf4j.MDC
+import kotlin.coroutines.ContinuationInterceptor
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+import kotlin.test.expect
+
+class SentryContextTest {
+
+    @BeforeTest
+    fun init() {
+        Sentry.init("https://key@sentry.io/123")
+    }
+
+    @AfterTest
+    fun close() {
+        Sentry.close()
+    }
+
+    @Test
+    fun testContextIsNotPassedByDefaultBetweenCoroutines() = runBlocking {
+        Sentry.setTag("myKey", "myValue")
+        // Standalone launch
+        GlobalScope.launch {
+            assertNull(getTag("myKey"))
+        }.join()
+    }
+
+    private fun getTag(tag:String): String? {
+        var value: String? = null
+        Sentry.configureScope {
+            value = it.tags[tag]
+        }
+        return value
+    }
+
+    @Test
+    fun testContextCanBePassedBetweenCoroutines() = runBlocking {
+        Sentry.setTag("myKey", "myValue")
+        // Scoped launch with MDCContext element
+        launch(SentryContext()) {
+            assertEquals("myValue", getTag("myKey"))
+        }.join()
+    }
+
+    @Test
+    fun testContextInheritanceMDC() = runBlocking {
+        MDC.put("myKey", "myValue")
+        withContext(MDCContext()) {
+            MDC.put("myKey", "myValue2")
+            // Scoped launch with inherited MDContext element
+            launch(Dispatchers.Default) {
+                assertEquals("myValue", MDC.get("myKey"))
+            }.join()
+        }
+        assertEquals("myValue", MDC.get("myKey"))
+    }
+
+    @Test
+    fun testContextInheritance() = runBlocking {
```

Comment:
This test is similar to `testContextInheritanceMDC` but uses `SentryContext`. It fails. @marandaneto perhaps you have an idea why?

Reply:
no idea just by looking at it, just wondering if we should actually clone the Hub or the Scope.
`Hub#clone` has never been used internally, we'd need to guarantee that its getting the right `stack`, something worth looking at